docker/build.sh

```diff
@@ -1,59 +1,48 @@
-#!/bin/sh -eux
+#!/bin/sh -eu
 
-# Placeholders to annotate the Github actions logs
-trace_on () { true; }
-trace_off () { true; }
-section_start () { true; }
-section_end () { true; }
+if [ "$#" -eq 0 ] || [ "$#" -gt 2 ]
+then
+	echo "Usage: $0 domjudge-version <namespace>"
+	echo "	For example: $0 5.3.0"
+	echo "	or: $0 5.3.0 otherNamespace"
+	exit 1
+fi
+
+# Placeholders for grouping log lines
+# (the body is a nested function declaration so it won't appear in the trace when using `set -x`)
+section_start() { _() { :; }; }
+section_end() { _() { :; }; }
 
 if [ -n "${CI+x}" ]
 then
 	if [ -n "${GITHUB_ACTION+x}" ]
-	set -x
 	then
-		# Functions to annotate the Github actions logs
-		trace_on () { set -x; }
-		trace_off () {
-			{ set +x; } 2>/dev/null
-		}
-		
-		section_start_internal () {
+		# Functions for grouping log lines on GitHub Actions
+		trace_on() { set -x; }
+		# trace_off is manually inlined so it won't appear in the trace
+		section_start() {
+			{ set +x; } 2>/dev/null # trace_off
 			echo "::group::$1"
 			trace_on
 		}
-		
-		section_end_internal () {
+		section_end() {
+			{ set +x; } 2>/dev/null # trace_off
 			echo "::endgroup::"
 			trace_on
 		}
-		
-		section_start () {
-			trace_off
-			section_start_internal "$@"
-		}
-		section_end () {
-			trace_off
-			section_end_internal
-		}
-	else
-		export PS4='(${0}:${LINENO}): - [$?] $ '
```

Comment:
Turns out that LINENO doesn't work in `sh`. How about switching to `bash`?

Reply:
We try by default ship `sh` scripts instead of `bash`, I don't think the `LINENO` is imporant enough to switch as this is only for CI and only in a legacy part of CI as this was used for gitlab-ci.

So in this case moving to `bash` is not worth it IMO, if you find another section where this makes sense I'm fine with (reconsider) moving. But to me the line numbers are not relevant when checking the CI log as this logging was only copied from the main repository but is not relevant for such a short script.

Reply:
I removed PS4 (and put the variable assignments in a log group to make things look nicer).